controllers/apps/operations/rebuild_instance.go

```diff
@@ -197,23 +230,284 @@ func (r rebuildInstanceOpsHandler) ReconcileAction(reqCtx intctrlutil.RequestCtx
+		progressDetail := r.getInstanceProgressDetail(*compStatus, instance.Name)
+		scalingOutPodName := r.getScalingOutPodNameFromMessage(progressDetail.Message)
+		pod := &corev1.Pod{}
+		if exist, err := intctrlutil.CheckResourceExists(reqCtx.Ctx, cli,
+			client.ObjectKey{Name: scalingOutPodName, Namespace: opsRes.Cluster.Namespace}, pod); err != nil {
+			return 0, 0, nil, err
+		} else if !exist {
+			reqCtx.Log.Info(fmt.Sprintf("waiting to create the pod %s", scalingOutPodName))
+			continue
+		}
+		isAvailable, err := instanceIsAvailable(synthesizedComp, pod, opsRes.OpsRequest.Annotations[ignoreRoleCheckAnnotationKey])
+		if err != nil {
+			// set progress status to failed when new pod is failed
+			failedCount += 1
+			completedCount += 1
+			progressDetail.SetStatusAndMessage(appsv1alpha1.FailedProgressStatus,
+				r.buildScalingOutPodMessage(scalingOutPodName, string(appsv1alpha1.UnavailablePhase)))
+			setComponentStatusProgressDetail(opsRes.Recorder, opsRes.OpsRequest, &compStatus.ProgressDetails, progressDetail)
+			continue
+		}
+		if !isAvailable {
+			// wait for the pod to be available
+			continue
+		}
+		if slices.Contains(compSpec.OfflineInstances, instance.Name) {
+			pod = &corev1.Pod{}
+			exist, err := intctrlutil.CheckResourceExists(reqCtx.Ctx, cli,
+				client.ObjectKey{Name: instance.Name, Namespace: opsRes.Cluster.Namespace}, pod)
+			if err != nil {
+				return 0, 0, nil, err
+			}
+			if !exist {
+				// f the pod that needs to be rebuilt is not found, and the new pod is available,
+				// it indicates that the rebuild process has been completed.
+				completedCount += 1
+				progressDetail.SetStatusAndMessage(appsv1alpha1.SucceedProgressStatus,
+					r.buildScalingOutPodMessage(scalingOutPodName, string(appsv1alpha1.AvailablePhase)))
+			} else {
+				progressDetail.SetStatusAndMessage(appsv1alpha1.ProcessingProgressStatus,
+					r.buildScalingOutPodMessage(scalingOutPodName, string(appsv1alpha1.AvailablePhase)))
+				if !pod.DeletionTimestamp.IsZero() && opsRes.OpsRequest.Force() {
+					// delete the pod forcibly
+					_ = intctrlutil.BackgroundDeleteObject(cli, reqCtx.Ctx, pod, client.GracePeriodSeconds(0))
+				}
+			}
+			setComponentStatusProgressDetail(opsRes.Recorder, opsRes.OpsRequest, &compStatus.ProgressDetails, progressDetail)
+		} else {
+			instancesNeedToOffline = append(instancesNeedToOffline, instance.Name)
+		}
+	}
+	return completedCount, failedCount, instancesNeedToOffline, nil
+}
+
+// offlineSpecifiedInstances to take the specific instances offline.
+func (r rebuildInstanceOpsHandler) offlineSpecifiedInstances(compSpec *appsv1alpha1.ClusterComponentSpec, clusterName string, instancesNeedToOffline []string) {
+	for _, insName := range instancesNeedToOffline {
+		compSpec.OfflineInstances = append(compSpec.OfflineInstances, insName)
+		templateName := appsv1alpha1.GetInstanceTemplateName(clusterName, compSpec.Name, insName)
+		if templateName == "" {
```

Comment:
How abount declare and replace "" (including other occurences) to a constant DefaultTemplateName